3rdparty/openssl/deprecations.h

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_OPENSSL_DEPRECATIONS_H
+#define _OE_OPENSSL_DEPRECATIONS_H
+
+#if !defined(OE_OPENSSL_SUPPRESS_DEPRECATIONS)
+
+#if defined(__cplusplus)
+#define OE_OPENSSL_EXTERN_C_BEGIN \
+    extern "C"                    \
+    {
+#define OE_OPENSSL_EXTERN_C_END }
+#else
+#define OE_OPENSSL_EXTERN_C_BEGIN
+#define OE_OPENSSL_EXTERN_C_END
+#endif
+
+#ifndef OE_DISABLE_OPENSSL_DEPRECATIONS
+#define OE_OPENSSL_DEPRECATED(MSG) __attribute__((deprecated(MSG)))
+#else
+#define OE_OPENSSL_DEPRECATED(MSG)
+#endif
+
+OE_OPENSSL_EXTERN_C_BEGIN
+
+#include <openssl/crypto.h>
```

Comment:
I'm guessing you're forcing the inclusion here so that the order in which the deprecations.h is included won't matter, but it does remind me of Anand's concern that the cmake/pkgconfig always pulls in the deprecation header for all object builds in the openSSL configuration, which means any module built targeting the openSSL configuration will include openSSL headers now (even if that particular component doesn't invoke openSSL).

I'm not sure there's a good way around that if we're not injecting this into the openSSL headers.

Reply:
Yes, the intention is forcing the order of inclusion because we're using the `-include` option and its order is not under our control. Is there any practice could walkaround this problem?

@anakrish

Reply:
This should be concern because we revert to the header patching approach.